app/controllers/documents_controller.rb

```diff
@@ -125,4 +134,37 @@ def prepare_for_mobile
     session[:mobile_param] = params[:mobile] if params[:mobile]
    # request.format = :mobile if mobile_device?
   end
+
+  private
+
+  def catalog_texts
+
+    if catalogue_enabled?
+       status, results = Melcatalog.texts
+       return results[:text] unless results[:text].nil?
+    end
+    return []
+  end
+
+  def catalog_content( doc )
+
+    if catalogue_enabled?
+      # we put placeholder content in earlier and replace with the real thing now
+      if doc.text.start_with?( "EID:" )
+         eid = doc.text.split( ":",2 )[ 1 ]
+         status, entry = Melcatalog.get( eid, 'stripxml' )
+         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
+           doc.text = entry[:text][ 0 ]['content']
+         else
+           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
+         end
+      end
+    end
+  end
+
+  # helper to determine if we should support content from the MEL catalog
+  def catalogue_enabled?
```

Comment:
Or just append to a single configuration... the widget names exist in code
anyway.

On Mon, Feb 23, 2015 at 3:37 PM, Jamie Folsom <notifications@github.com>
wrote:

> In app/controllers/documents_controller.rb
> <https://github.com/hyperstudio/Annotation-Studio/pull/128#discussion_r25199472>
> :
>
> > +    if catalogue_enabled?
> > +      # we put placeholder content in earlier and replace with the real thing now
> > +      if doc.text.start_with?( "EID:" )
> > +         eid = doc.text.split( ":",2 )[ 1 ]
> > +         status, entry = Melcatalog.get( eid, 'stripxml' )
> > +         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
> > +           doc.text = entry[:text][ 0 ]['content']
> > +         else
> > +           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
> > +         end
> > +      end
> > +    end
> > +  end
> > +
> > +  # helper to determine if we should support content from the MEL catalog
> > +  def catalogue_enabled?
>
> On second thought, there is a reasonably easy, if rather kludgy, solution,
> by adding the following to the annotator setup:
>
>   var optionsRichText = {
>     editor_enabled: '<%= ENV["ANNOTATOR_RICHTEXT"] %>',
>     tinymce: {
>       <% if ENV["CATALOG_ENABLED"] == 'true' %>
>       'toolbar': '<%= ENV["ANNOTATOR_RICHTEXT_WITH_CATALOG"] %>',
>       <% end %>
>       'toolbar': '<%= ENV["ANNOTATOR_RICHTEXT_CONFIG"] %>',
>       'image_dimensions': false
>     }
>   };
>
> And creating alternate config vars for tinymce.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hyperstudio/Annotation-Studio/pull/128/files#r25199472>
> .
>